x-pack/test/functional/apps/maps/import_geojson/add_layer_import_panel.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import path from 'path';
+
+export default function ({ getPageObjects }) {
+  const PageObjects = getPageObjects(['maps', 'common']);
+
+  const IMPORT_FILE_PREVIEW_NAME = 'Import File';
+  const FILE_LOAD_DIR = 'test_upload_files';
+  const DEFAULT_LOAD_FILE_NAME = 'point.json';
+  const GEO_POINT = 'geo_point';
+  const GEO_SHAPE = 'geo_shape';
+
+  describe('Import layer panel', () => {
+    before(async () => {
+      await PageObjects.maps.openNewMap();
+    });
+
+    beforeEach(async () => {
+      await PageObjects.maps.clickAddLayer();
+      const panelOpen = await PageObjects.maps.isLayerAddPanelOpen();
+      expect(panelOpen).to.be(true);
+
+      await PageObjects.maps.selectGeoJsonUploadSource();
+      await PageObjects.maps.uploadJsonFileForIndexing(
+        path.join(__dirname, FILE_LOAD_DIR, DEFAULT_LOAD_FILE_NAME)
+      );
+      await PageObjects.maps.waitForLayersToLoad();
```

Comment:
Move `await PageObjects.maps.waitForLayersToLoad()` into `uploadJsonFileForIndexing` since waiting for the layer to load is part of uploading the file. Then you can remove all of the calls to waitForLayersToLoad after each call to uploadJsonFileForIndexing